app/src/main/java/com/example/drawingactivity/CustomColorViewModel.kt

```diff
@@ -0,0 +1,113 @@
+package com.example.drawingactivity
+
+import android.graphics.Color
+
+/*
+What do I need to do?
+Tell the Gradient to change Hues when the slider is changed
+Tell the Textviews to change value when the user picks a hue
+Hexcode, RGB, and HSV
+Update the Slider then the Textviews change Hue
+Update the ColorBlocks when new colors are picked
+Save the state of all of the values
+
+Make a Color Block
+Holds a color in a rectangle
+matches the exact value the user put in
+BONUS: Draw the Hexcode on the color for easier ID.
+*/
+
+class CustomColorViewModel(
+
+) {
+
+    // this lambda is used to invoke a function that will update all views.
+    // We give it the values in the Viewstates and it does not return anything, it simply updates
+    var onUpdate: ((ViewStates) -> Unit)? = null
+
+    data class ViewStates(
+        var hue: Int,
+        var sat: Float,
+        var bright: Float,
+        var red: Int,
+        var green: Int,
+        var blue: Int,
+        var primaryColor: Int,
+    )
+
+    private var hue = 0
```

Comment:
All of this isn't part of this ticket. Let's defer all of this and just go one ticket at a time

Reply:
deleted